source/Coop/Mod/Managers/ClientManager.cs

```diff
@@ -1,51 +1,41 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Security.Permissions;
-using System.Text;
-using System.Threading.Tasks;
-using Coop.Mod.Serializers;
 using SandBox;
-using StoryMode;
-using StoryMode.CharacterCreationSystem;
 using TaleWorlds.CampaignSystem;
 using TaleWorlds.Core;
-using TaleWorlds.Engine.Screens;
-using TaleWorlds.MountAndBlade;
-using TaleWorlds.ObjectSystem;
 using TaleWorlds.SaveSystem.Load;
 using System.Reflection;
-using NetworkMessages.FromClient;
-using Module = TaleWorlds.MountAndBlade.Module;
 using TaleWorlds.CampaignSystem.Actions;
-using TaleWorlds.Library;
-using TaleWorlds.Localization;
-using Helpers;
+using System.Threading.Tasks;
 
 namespace Coop.Mod.Managers
 {
     public class ClientManager : CampaignGameManager
     {
-        public ClientManager(LoadResult saveGameData) : base(saveGameData) { }
+        readonly string m_PartyName;
+        Hero clientPlayer;
+        public ClientManager(LoadResult saveGameData, string partyName) : base(saveGameData) { m_PartyName = partyName; }
 
         public delegate void OnOnLoadFinishedEventHandler(object source, EventArgs e);
         public static event OnOnLoadFinishedEventHandler OnLoadFinishedEvent;
-
         public override void OnLoadFinished()
         {
             base.OnLoadFinished();
-
+            Parallel.ForEach(MobileParty.All, (party) =>

```

Comment:
Feels kinda weird to express it that way. The code is not actually thread save, it only works under the assumption that only 1 party.Name == m_PartyName.

How about this instead:
```C#
MobileParty playerParty = MobileParty.All.AsParallel().SingleOrDefault(party => party.Name.ToString() == m_PartyName);
if(playerParty != null)
{
   // ...
}
```

Reply:
I agree